integration/oplog/update_oplog_test.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oplog
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/FerretDB/FerretDB/integration"
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+func TestOplogUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx, coll := setup.Setup(t, shareddata.Composites)
+	local := coll.Database().Client().Database("local")
+	ns := fmt.Sprintf("%s.%s", coll.Database().Name(), coll.Name())
+	opts := options.FindOne().SetSort(bson.D{{"$natural", -1}})
+
+	err := local.CreateCollection(ctx, "oplog.rs", options.CreateCollection().SetCapped(true).SetSizeInBytes(536870912))
+	if err != nil {
+		require.Contains(t, err.Error(), "local.oplog.rs already exists")
+		err = nil
+	}
+
+	require.NoError(t, err)
+
+	for name, tc := range map[string]struct { //nolint:vet // for readability
+		update         bson.D
+		filter         bson.D
+		expectedDiffV1 *types.Document
+		expectedO2     *types.Document
```

Comment:
What is the meaning of O2?

Reply:
o2 only exists for diff (o) with $v: 1 and contains the id of the oploged entry. At least, as far as I understand :)